engine/utils/collections/Iterators.h

```diff
@@ -0,0 +1,200 @@
+//
+//
+//  Copyright (c) 2022 Jonathan Moallem (@J-Mo63) & Aryeh Zinn (@Raelr)
+//
+//  This code is released under an unmodified zlib license.
+//  For conditions of distribution and use, please see:
+//      https://opensource.org/licenses/Zlib
+//
+
+#ifndef SIEGE_ENGINE_ITERATORS_H
+#define SIEGE_ENGINE_ITERATORS_H
+
+#include <cstddef>
+namespace Siege::Utils
+{
+/**
+ * A standard linear iterator. This iterator always iterates over every element of the array
+ * regardless of whether it has been filled or not. This is the 'standard' iterator type
+ * @tparam T the type of array stored by the iterator. Must be an array type which exposes a
+ * 'Data()' and `Size()` function
+ * @tparam R The type returned by the iterator (usually the type of the array)
+ */
+template<typename T, typename R>
```

Comment:
I think it's usually `A` and `T` in this case so `T` continues to refer to the type held by the array.

Reply:
Makes sense, Have changed accordingly.